src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4682,6 +4678,17 @@ static TypeWithState calculateResultType(TypeWithAnnotations targetTypeWithNulla
             return operandType;
         }
 
+        private void CheckpointWalkerThroughConversionGroup(BoundConversion conversionOpt, BoundExpression convertedNode)
```

Comment:
I chose to add this as a separate method instead of putting it in `RemoveConversion` directly, as `RemoveConversion` is often used without then visiting the underlying expression. There are tradeoffs to both though, so there's preference on adding it to `RemoveConversion`, controlled by a default parameter, I can do that as well.